exist-core/src/main/resources/org/exist/xquery/lib/xqsuite/xqsuite.xql

```diff
@@ -801,7 +801,7 @@ declare %private function test:assertEqualsPermutation($values as element(value)
             let $tests :=
                 for $item in $result
                 return
-                    some $value in $values satisfies test:equals($item, $value)
+                    some $value in $values satisfies test:equals($value, $item)
```

Comment:
Good catch! We should consider renaming `$item` to `$result` in this function; `$result` to `$results`; and perhaps even `$value` and `$values` to `$annotation-value` and `$annotation-values`. This would make it quite apparent how these variables need to align with the `test:equals()` function signature: `test:equals($annotation-value as element(value), $result as item())`. (Perhaps in a future PR, that is.)

Reply:
I would be in favour of changing the name as part of this PR